File: gfg/linked-list/set_1.java
class Node {
  int data;
  Node next;

  Node(int data) {
    this.data = data;
    this.next = null;
  }
}


class LinkedList {
  Node head;

  LinkedList() {
    this.head = null;
  }

  void printList() {
    Node temp = this.head;
    while (temp != null) {
      System.out.println("" + temp.data + " ");
      temp = temp.next;
    }
    System.out.println();
  }
}


public class set_1 {

  public static void main(String args[]) {

    LinkedList list = new LinkedList();
    Node head = new Node(1);
    Node second = new Node(2);
    Node third = new Node(3);
    head.next = second;
    second.next = third;

    list.head = head;

    list.printList();
  }
}

File: gfg/linked-list/set_1.cpp
#include <cstdio>
#include <cstdlib>

struct Node {
  int data;
  struct Node *next;
};

void printList(struct Node *head) {

  while (head) {
    printf("%d ", head->data);
    head = head->next;
  }
  printf("\n");
}

int main() {

  struct Node *head = (struct Node *) malloc(sizeof(struct Node));
  struct Node *second = (struct Node *) malloc(sizeof(struct Node));
  struct Node *third = (struct Node *) malloc(sizeof(struct Node));

  head->data = 1;
  head->next = second;
  second->data = 2;
  second->next = third;
  third->data = 3;
  third->next = nullptr;

  printList(head);

  return 0;
}
